### PR TITLE
Github actions improvements

### PR DESCRIPTION
**Pull request by hbkse**
Adds a section for linting and fixes the integration tests run

### CODE REVIEW

File: .github/workflows/build_test.yml [hunk 1]
```diff
@@ -15,22 +15,38 @@ on:
...
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
+      uses: actions/setup-node@v3
       with:
         node-version: ${{ matrix.node-version }}
     - run: npm ci
     - run: npm run build --if-present
+    - run: node build_release_package_json.js
```
Comment (kbridbur) [inline .github/workflows/build_test.yml]: Is this needed for unit tests?

File: .github/workflows/build_test.yml [hunk 2]
```diff
@@ -15,22 +15,38 @@ on:
...
+        node-version: [8.x, 10.x, 12.x, 14.x, 16.x, 17.x]
 
+    steps:
+    - uses: actions/checkout@v3
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v3
+      with:
+        node-version: ${{ matrix.node-version }}
+    - run: npm ci
+    - run: npm run build --if-present
```
Comment (kbridbur) [inline .github/workflows/build_test.yml]: Is this needed to lint?

**Review by kbridbur — COMMENTED** [review]
lgtm

File: .github/workflows/build_test.yml [hunk 3]
```diff
@@ -15,22 +15,38 @@ on:
...
     strategy:
       matrix:
         node-version: [8.x, 10.x, 12.x, 14.x, 16.x, 17.x]
 
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v3
     - name: Use Node.js ${{ matrix.node-version }}
-      uses: actions/setup-node@v2
+      uses: actions/setup-node@v3
```
Comment (amikofalvy) [inline .github/workflows/build_test.yml:43]: Did you check on the changelog for v3? What do we get with this? are there breaking changes?

**Review by amikofalvy — COMMENTED** [review]
Please strip out any unnecessary actions and steps as kyle suggests. Then ship it

File: .github/workflows/build_test.yml (hunk 3 above)
Comment (hbkse) [reply .github/workflows/build_test.yml:43]: It upgrades the runtime nodejs and removes some deprecated tags, should not be breaking for us

**Review by kbridbur — APPROVED** [review]

**Pull request merged**